src/braket/aws/aws_quantum_task_batch.py

```diff
@@ -0,0 +1,231 @@
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+from __future__ import annotations
+
+import time
+from concurrent.futures.thread import ThreadPoolExecutor
+from typing import List, Set, Union
+
+from braket.annealing import Problem
+from braket.aws.aws_quantum_task import AwsQuantumTask
+from braket.aws.aws_session import AwsSession
+from braket.circuits import Circuit
+
+
+class AwsQuantumTaskBatch:
+    """Executes and a batch of quantum tasks in parallel.
+
+    Using this class can yield vast speedups over executing tasks sequentially,
+    and is particularly useful for computations that can be parallelized,
+    such as calculating quantum gradients or statistics of terms in a Hamiltonian.
+
+    Note: there is no benefit to using this method with QPUs outside of their execution windows,
+    since results will not be available until the window opens.
+    """
+
+    MAX_WORKERS = 100
+    MAX_PARALLEL_DEFAULT = 10
+    TERMINAL_STATES = AwsQuantumTask.NO_RESULT_TERMINAL_STATES.union(
+        AwsQuantumTask.RESULTS_READY_STATES
+    )
+
+    def __init__(
+        self,
+        aws_session: AwsSession,
+        device_arn: str,
+        task_specifications: List[Union[Circuit, Problem]],
+        s3_destination_folder: AwsSession.S3DestinationFolder,
+        shots: int,
+        max_parallel: int = MAX_PARALLEL_DEFAULT,
+        poll_interval_seconds: float = AwsQuantumTask.DEFAULT_RESULTS_POLL_INTERVAL,
+        *aws_quantum_task_args,
+        **aws_quantum_task_kwargs,
+    ):
+        """Creates a batch of quantum tasks.
+
+        Args:
+            aws_session (AwsSession): AwsSession to connect to AWS with.
+            device_arn (str): The ARN of the quantum device.
+            task_specification (Union[Circuit, Problem]): The specification of the task
+                to run on device.
+            s3_destination_folder (AwsSession.S3DestinationFolder): NamedTuple, with bucket
+                for index 0 and key for index 1, that specifies the Amazon S3 bucket and folder
+                to store task results in.
+            shots (int): The number of times to run the task on the device. If the device is a
+                simulator, this implies the state is sampled N times, where N = `shots`.
+                `shots=0` is only available on simulators and means that the simulator
+                will compute the exact results based on the task specification.
+            max_parallel (int): The maximum number of tasks to run on AWS in parallel.
+                Batch creation will fail if this value is greater than the maximum allowed
+                concurrent tasks on the device. Default: 10
+            poll_interval_seconds (float): The polling interval for result(), default is 0.25
```

Comment:
Nit: I'd remove the description for the default. Otherwise if we decide to update the default later we'd have to track down every comment we mentioned it in.

Reply:
Customers always want to know the default.

Reply:
The default is `AwsQuantumTask.DEFAULT_RESULTS_POLL_INTERVAL`, as can be seen from the signature. I'd rather mention that as the default instead of 0.25 seconds.